### PR TITLE
Ban import path to react and mui in the background script with a new linter

### PR DESCRIPTION
**Pull request by Jack-Works**

### CODE REVIEW

**Comment by neruthes** [comment]
Every pull request should close at least 1 issue. Remember always create an issue before implementing a pull request.

File: scripts/import-linter.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        if (x.includes('@material-ui')) {
+            return `@material-ui appears in the import chain!
+To write a cross context file, use safeMUI() inside the function to import them conditionally.
+`
+        } else if (x === 'react') {
+            return `React appears in the import chain!
+To write a cross context file, use safeReact() inside the function to import them conditionally.
+`
+        }
+        return null
```
Comment (guanbinrui) [inline scripts/import-linter.ts]: `null` is not asignable to string type.

File: src/utils/safeRequire.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    return require('@material-ui/core') as typeof import('@material-ui/core')
+}
+
+export function safeGetActiveUI() {
+    if (GetContext() === 'background') throw new Error('Illegal context')
+    return (require('../social-network/ui') as typeof import('../social-network/ui')).getActivatedUI()
+}
+
+export function safeOptionsPageWorker() {
+    if (GetContext() !== 'options') throw new Error('Illegal context')
```
Comment (SunriseFox) [inline src/utils/safeRequire.ts]: Seems the background page requires it to work. Without `require('./provider.worker')` it cannot decrypt posts...

File: src/utils/i18n.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import en from '../_locales/en/messages.json'
 import zh from '../_locales/zh/messages.json'
-import React from 'react'
 import { GetContext } from '@holoflows/kit/es'
+import { safeReact, safeGetActiveUI } from './safeRequire.js'
```
Comment (SunriseFox) [inline src/utils/i18n.ts]: should it be './safeRequire(.ts)'?

File: scripts/import-linter.ts (hunk 1 above)
Comment (Jack-Works) [reply scripts/import-linter.ts]: yeah you are right that file is in the scripts/ folder so the strict null check is not opened

File: src/utils/i18n.ts (hunk 3 above)
Comment (Jack-Works) [reply src/utils/i18n.ts]: Oh that's strange. Import .ts file with a .js extname is valid in typescript. It must be there're some wrong configuration in our webpack config.

**Comment by Jack-Works** [comment]
@guanbinrui hello?

**Pull request merged**